src/app/public/modules/country-field/country-field.component.ts

```diff
@@ -0,0 +1,317 @@
+
+      this.defaultCountryData = this.countries
+        .find(country => country.iso2 === this._defaultCountry);
+
+      this.sortCountriesWithSelectedAndDefault(this.selectedCountry);
+    }
+  }
+
+  public get defaultCountry(): string {
+    return this._defaultCountry;
+  }
+
+  /**
+   * Indicates whether to disable the country field.
+   */
+  @Input()
+  public disabled: boolean = false;
+
+  /**
+   * Fires when the selected country changes.
+   */
+  @Output()
+  public selectedCountryChange: EventEmitter<SkyCountryFieldCountry> = new EventEmitter<SkyCountryFieldCountry>();
+
+  public countries: SkyCountryFieldCountry[];
+
+  public countrySearchFormControl: FormControl;
+
+  public isInPhoneField: boolean = false;
+
+  public isInputFocused: boolean = false;
+
+  public searchTextMinimumCharacters: number = 2;
+
+  @ViewChild(SkyAutocompleteInputDirective)
+  public countrySearchAutocompleteDirective: SkyAutocompleteInputDirective;
+
+  public set selectedCountry(newCountry: SkyCountryFieldCountry) {
+    if (this._selectedCountry !== newCountry) {
+      this._selectedCountry = newCountry;
+
+      this.sortCountriesWithSelectedAndDefault(newCountry);
+
+      this.countrySearchFormControl.setValue(this.selectedCountry);
+
+      if (!this.isFirstChange) {
+        this.onChange(newCountry);
+        this.onTouched();
+
+        this.selectedCountryChange.emit(newCountry);
+      }
+
+      // Do not mark the field as "dirty"
+      // if the field has been initialized with a value.
+      if (this.isFirstChange && this.control) {
+        this.control.markAsPristine();
+      }
+
+      if (this.isFirstChange && newCountry !== null) {
+        this.isFirstChange = false;
+      }
+    }
+  }
+
+  public get selectedCountry(): SkyCountryFieldCountry {
+    return this._selectedCountry;
+  }
+
+  private control: AbstractControl;
+
+  private defaultCountryData: SkyCountryFieldCountry;
+
+  private idle: Subject<any> = new Subject();
+
+  private isFirstChange: boolean = true;
+
+  private ngUnsubscribe = new Subject();
+
+  private _defaultCountry: string;
+
+  private _selectedCountry: SkyCountryFieldCountry;
+
+  constructor(
+    private changeDetector: ChangeDetectorRef,
+    private elRef: ElementRef,
+    private windowRef: SkyAppWindowRef
+  ) {
+    /**
+     * The json functions here ensures that we get a copy of the array and not the global original.
+     * This ensures that multiple instances of the component don't overwrite the original data.
+     *
+     * We must type the window object as any here as the intl-tel-input library adds its object
+     * to the main window object.
+     */
+    this.countries = JSON.parse(JSON.stringify((window as any)
+      .intlTelInputGlobals.getCountryData()));
+
+    this.countrySearchFormControl = new FormControl();
+
+    this.isInPhoneField = (<HTMLElement>elRef.nativeElement.parentElement)
+      .classList
+      .contains('sky-phone-field-country-search');
```

Comment:
We may have already talked about this, but is there no way to get the parent by using a `@Optional` or `@Host` injector in the constructor args? This logic seems brittle... what if later down the line, we slide in a couple dom elements between the country field and phone-field? Is there a better way?

Reply:
Nope...... we can't use `@Host` because it would introduce a peer dependency of the phone field library into lookup (and phone field already has a peer on lookup). I don't love it either but the only way to make this less brittle is to traverse the entire dom upwards and I hated doing that when I know that the country field will go in this exact spot when I covert phone field. It is brittle though.... IDK..... thoughts?

Reply:
Talked offline. While its a little brittle, this phone field is the only scenario in which the country field will have different behavior. Since we control that interaction, we're ok with this implementation for now. In the future, if we add more unique behavior like this, it might make sense to implement a `type` that can handle these types of settings.